server/repocreds/repocreds.go

```diff
@@ -65,6 +65,30 @@ func (s *Server) ListRepositoryCredentials(ctx context.Context, q *repocredspkg.
 	return &appsv1.RepoCredsList{Items: items}, nil
 }
 
+// ListWriteRepositoryCredentials returns a list of all configured repository credential sets
+func (s *Server) ListWriteRepositoryCredentials(ctx context.Context, q *repocredspkg.RepoCredsQuery) (*appsv1.RepoCredsList, error) {
+	urls, err := s.db.ListRepositoryCredentials(ctx)
+	if err != nil {
+		return nil, err
+	}
+	items := make([]appsv1.RepoCreds, 0)
+	for _, url := range urls {
+		if s.enf.Enforce(ctx.Value("claims"), rbacpolicy.ResourceWriteRepositories, rbacpolicy.ActionGet, url) {
```

Comment:
write repo is not bounded by project, then how enforcer is used? sorry if my question is vague

Reply:
Even if it's not project-scoped, the user still must have RBAC access to get the write-repo.